apps/api/src/app/integrations/usecases/get-active-integration/get-active-integration.usecase.ts

```diff
@@ -1,20 +1,61 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
 
-import { GetIntegrationsCommand } from '../get-integrations/get-integrations.command';
+import { SelectIntegration, SelectIntegrationCommand } from '@novu/application-generic';
+import { ChannelTypeEnum } from '@novu/shared';
+import { IntegrationEntity, IntegrationRepository } from '@novu/dal';
+
+import { GetActiveIntegrationsCommand } from './get-active-integration.command';
+import { GetActiveIntegrationResponseDto } from '../../dtos/get-active-integration-response.dto';
 
 @Injectable()
 export class GetActiveIntegrations {
-  constructor(private getDecryptedIntegrationsUsecase: GetDecryptedIntegrations) {}
-
-  async execute(command: GetIntegrationsCommand): Promise<IntegrationEntity[]> {
-    return await this.getDecryptedIntegrationsUsecase.execute(
-      GetDecryptedIntegrationsCommand.create({
-        organizationId: command.organizationId,
-        active: true,
-        userId: command.userId,
-      })
-    );
+  constructor(private integrationRepository: IntegrationRepository, private selectIntegration: SelectIntegration) {}
+
+  async execute(command: GetActiveIntegrationsCommand): Promise<GetActiveIntegrationResponseDto[]> {
+    const channelTypes = Object.values(ChannelTypeEnum);
+
+    const activeIntegration = await this.integrationRepository.find({
+      _organizationId: command.organizationId,
+      _environmentId: command.environmentId,
+      active: true,
+    });
+
+    if (!activeIntegration.length) {
+      throw new NotFoundException(`No active integrations found for environment ${command.environmentId}`);
+    }
+
+    const selectedIntegrations = await this.getSelectedIntegrations(channelTypes, command);
+
+    return this.mapBySelectedIntegration(activeIntegration, selectedIntegrations);
   }
+
+  private mapBySelectedIntegration(activeIntegration: IntegrationEntity[], selectedIntegrations: IntegrationEntity[]) {
+    return activeIntegration.map((integration) => {
+      const selected = selectedIntegrations.find((selectedIntegration) => selectedIntegration._id === integration._id);
+
+      return selected ? { ...integration, selected: true } : { ...integration, selected: false };
+    });
+  }
+
+  private async getSelectedIntegrations(channelTypes: string[], command: GetActiveIntegrationsCommand) {
+    return (
+      await Promise.all(
+        channelTypes.map(async (channelType) => {
+          return await this.selectIntegration.execute(
+            SelectIntegrationCommand.create({
+              organizationId: command.organizationId,
+              environmentId: command.environmentId,
+              userId: command.userId,
+              channelType: channelType as ChannelTypeEnum,
+              providerId: command.providerId,
+            })
+          );
+        })
+      )
+    ).filter(notEmpty);
+  }
+}
+
+export function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
```

Comment:
Not particularly fond of the name: [] is empty, we have empty string.
The behaviour we try to implement is similar to Nullish Coalescing, so I'd suggest to rename it to `notNullish`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

Reply:
tend to agree, I am good with notNullish :)